lib/surface/compiler.ex

```diff
@@ -574,11 +577,15 @@ defmodule Surface.Compiler do
     ]
   end
 
+  # TODO: [Type] Move this logic to TypeHandler
```

Comment:
The question is: should we just move the code as it is to `Surface.TypeHandler` or would it be better to delegate to each type the ability to match its own type? Something like:

```elixir
defmodule Boolean do
  use SurfaceTypeHandler
  ...
  @impl true
  def match_type?(nil, name, _meta) do
    name in @boolean_tag_attributes
  end
end
```

I'll be fine by just moving the code to `Surface.TypeHandler`. I was just wondering if it would be the best approach.

Reply:
Let's just leave this one as it is. This is just about the compiler detecting the types, not handling them.